File: Desenvolvimento/client/src/@fuse/types/models/logradouro.ts
import { Pais } from './pais';
import { Estado } from './estado';
import { Municipio } from './municipio';
import { BaseModel } from '../base-model';

export class Logradouro extends BaseModel {
    constructor(
        public id: number,
        public cep?: string,
        public endereco?: string,
        public bairro?: string,
        
        public paisId?: number,
        public pais?: Pais, 
        
        public estadoId?: number,
        public estado?: Estado,

        public municipioId?: number,
        public municipio?: Municipio 
    ) { super(); }
}

File: Desenvolvimento/server/src/ApplicationCore/Entities/Logradouro.cs
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class Logradouro : IBaseEntity<long>
    {
        public long Id { get; set; }
        public string CEP { get; set; }
        public string Endereco { get; set; }
        public string Bairro { get; set; }
        
        public long? PaisId { get; set; }
        public virtual Pais Pais { get; set; }

        public long? EstadoId { get; set; }
        public virtual Estado Estado { get; set; }

        public long? MunicipioId { get; set; }
        public virtual Municipio Municipio  { get; set; }
    }
}
